src/gennylib/src/context.cpp

```diff
@@ -124,16 +123,18 @@ std::unordered_map<PhaseNumber, std::unique_ptr<PhaseContext>> ActorContext::con
                   "Every phase should have at least be an empty map.";
             throw InvalidConfigurationException(ss.str());
         }
-
-        auto configuredIndex = phase["Phase"].as<PhaseNumber>(index);
-        auto [it, success] =
-            out.try_emplace(configuredIndex, std::make_unique<PhaseContext>(phase, *actorContext));
-        if (!success) {
-            std::stringstream msg;
-            msg << "Duplicate phase " << configuredIndex;
-            throw InvalidConfigurationException(msg.str());
+        PhaseRangeSpec configuredRange =
+            phase["Phase"].as<PhaseRangeSpec>(PhaseRangeSpec{IntegerSpec{totalIndex}});
+        for (int rangeIndex = configuredRange.start; rangeIndex <= configuredRange.end; rangeIndex++) {
+            auto [it, success] =
+                out.try_emplace(rangeIndex, std::make_unique<PhaseContext>(phase, *actorContext));
+            if (!success) {
+                std::stringstream msg;
+                msg << "Duplicate phase " << rangeIndex;
+                throw InvalidConfigurationException(msg.str());
+            }
+            totalIndex++;
```

Comment:
suggestion - rename `totalIndex` to `lastPhaseNumber` since that's what it's tracking